File: Bit Magic.cpp
class Solution {
  public:
    int bitMagic(int n, vector<int> &arr) {
        // code here
        int i = 0 , j = n-1;
        int cnt = 0;
        
        while(i < j) {
            if(arr[i] != arr[j]) {
                cnt++;
            }
            i++;
            j--;
        }
        return cnt % 2 == 0 ? cnt/2 : cnt/2+1;
    }
};

File: Bit Magic.java
class Solution {
    public static int bitMagic(int n, int[] arr) {
        // code here
        int op=0;
        for(int i=0; i<n/2; i++){
            if(arr[i]!= arr[n-1-i]){
                op++;
            }
        }
        return (op+1)/2;
    }
}
        
